packages/components/src/server/hive-layout.tsx

```diff
@@ -0,0 +1,142 @@
+import { DetailedHTMLProps, HtmlHTMLAttributes, ReactElement, ReactNode } from 'react';
+import { Layout } from 'nextra-theme-docs';
+import { Head } from 'nextra/components';
+import { getPageMap } from 'nextra/page-map';
+import { cn } from '../cn';
+import { Body, BodyProps } from './body.client';
+
+export interface HiveLayoutProps
+  extends DetailedHTMLProps<HtmlHTMLAttributes<HTMLHtmlElement>, HTMLHtmlElement> {
+  children: ReactNode;
+  head: ReactNode;
+  navbar: ReactElement;
+  footer: ReactElement;
+  fontFamily: string;
+  lightOnlyPages: string[];
+  bodyProps?: DetailedHTMLProps<HtmlHTMLAttributes<HTMLBodyElement>, HTMLBodyElement>;
+}
+
+/**
+ * Alternative to `GuildLayout` for Hive-branded websites.
+ *
+ * Accepts navbar and footer as slots/children props, because they're highly customizable,
+ * and their defaults belong to HiveNavigation and HiveFooter component default props.
+ *
+ * ## Configuration
+ *
+ * Pages can differ by widths and supported color schemes:
+ *
+ * - The footer in docs has 90rem width, in landing pages it has 75rem.
+ * - The navbar in docs has 90rem width, in landing pages it has 1392px.
+ * - Landing pages only support light mode for _business and prioritization reasons_.
+ *
+ * TODO: Consider unifying this in design phase.
+ *
+ * For now, a page or a layout can configue these as follows:
+ *
+ * ### Light-only pages
+ *
+ * @example
+ * ```tsx
+ * <HiveLayout bodyProps={{ lightOnlyPages: ['/', '/friends'] }} />
+ * ```
+ *
+ * This will force light theme to the pages with paths `/` and `/friends`,
+ * by adding `.light` class to the <body /> element.
+ *
+ * ### Landing page widths
+ *
+ * @example
+ * ```tsx
+ * import { HiveLayoutConfig } from '@theguild/components'
+ *
+ * <HiveLayoutConfig widths="landing-narrow" />
```

Comment:
You can put this in a website root layout based on the same include list as `lightOnlyPages`, but it's IMHO even better to put it in a deeper layout or even pages itself (it renders just one div with id).